### ISSUE TITLE
Stack overflow error with responsive heatmap and coloraxis in styled build

### ISSUE THREAD

**Issue by RainerSchwarze** [highcharts/highcharts#10477]
#### Expected behaviour

Setting responsive options for heatmap with color axis; then resizing browser width does not cause errors (in console).

#### Actual behaviour

Setting responsive options for heatmap with color axis; then resizing browser width causes errors (in console).

#### Live demo with steps to reproduce
https://jsfiddle.net/qeov5rjh/

- increase browser width
- decrease until legend jumps below chart
- increase browser width again until the legend disappears
- observe console output

I basically turned the heatmap demo to use the styled build and added these options:

```
    responsive: {
        rules: [{
            condition: {
                maxWidth: 500
            },
            chartOptions: {
                legend: {
                    align: 'center',
                    verticalAlign: 'bottom',
                    layout: 'horizontal'
                }
            }
        }]
    }
```

#### Product version
Latest version

#### Affected browser(s)
tested with Chrome

#### Console error:

```
highcharts.src.js:885 Uncaught RangeError: Maximum call stack size exceeded
    at x (highcharts.src.js:885)
    at Object.<anonymous> (highcharts.src.js:899)
    at Object.a.objectEach (highcharts.src.js:2308)
    at x (highcharts.src.js:891)
    at Object.<anonymous> (highcharts.src.js:899)
    at Object.a.objectEach (highcharts.src.js:2308)
    at x (highcharts.src.js:891)
    at a.merge (highcharts.src.js:919)
    at b.ColorAxis.b.ColorAxis.update (highcharts.src.js:34265)
    at b.ColorAxis.b.ColorAxis.update (heatmap.src.js:855)
```

From a first look it seems an infinite recursion with colorAxis updates.

**Comment by TorsteinHonsi** [comment]
Simpler demo with reliable button: https://jsfiddle.net/ejtsuqaf/

**Comment by TorsteinHonsi** [comment]
Turns out the issue is fixed: https://jsfiddle.net/ejtsuqaf/1/

The demo used the old styled mode pre-built files, that are deprecated and not being updated since v7.0.0.

**Comment by RainerSchwarze** [comment]
Do you know, how one could fix this in v6?
Would the fix for #9617 be related to this?
